### DISCUSSION TITLE
Convergence failure

### DISCUSSION THREAD

**Discussion by prsdshiva** [neperfepx/FEPX#39] [Q&A] [answered]
Hello,

I am attempting to run a model similar to the "1_uniaxial" example that comes with the package. I have changed the material properties to that of steel and the model dimension is set to 0.5 X 0.5 X 2 mm, containing roughly 12k grains (approx 900k elements).  I am encountering following error while running this model. 

 `Info . Solving NR iteration... ERROR  :       . CG_SOLVER_EBE: Convergence failure.`

Any help would be highly appreciated. Attaching the error file here with.
[fepx_example_out.txt](https://github.com/acmelab-ua/FEPX/files/7001398/fepx_example_out.txt)

Regards
Shiva

**Comment by mkasemer** [comment, marked as answer]
Hello Shiva,

Thanks for interest in FEPX. Unfortunately, I can't diagnose the problem based on what you have given. At the least, you must provide a [minimum working example ](https://stackoverflow.com/help/minimal-reproducible-example).

However, I suspect that this simulation may be too large, or have ill-formed elements, which are leading to premature failure. I'm not sure this is easily diagnosable, as it is probably a function of the mesh, which is very large. How are you producing the mesh?

**Edit:** I should be more clear about my response. It is possible that with a large mesh, there are some elements which may be poorly formed, which consequently may cause poor convergence behavior in the FE solver. Often times, this happens when entering the plastic regime, and elemental deformation may become more localized. It is also possible that the strain steps are too large. The examples provided with FEPX take relatively large strain steps in an effort to reduce simulation time (i.e., allow for quick demonstrations of capabilities). Your failure could very well be a function of large strain steps, and simply taking intermediary steps will allow for the solver to properly converge. The fact that your simulation is failing at the point it does could indicate either issue: it is both at the point when plastic deformation is likely beginning to take off, and it is also when you take a relatively large strain step (compared to the previous steps).

Might I also suggest: Attempt exploratory simulations on smaller meshes, either using a lower number of grains, a lower number of elements per grain, or both. Running a simulation using a mesh of ~1M elements on only 24 cores must be rather slow, and it is best to "work out the bugs" in your simulations on smaller, more tractable meshes, and running the "big" simulations only when you know you are ready.

**Reply by prsdshiva** [reply]
Thank you @mkasemer for your reply. I am relatively new to FEPX and have been exploring the examples. 

I will look into the strain steps/mesh quality and try out a relatively smaller model before moving to "big" scale simulations. If I still encounter problems, I will get back to you with a working example.
